### ISSUE TITLE
Problem with public state variable overriding function

### ISSUE THREAD

**Issue by chriseth** [ethereum/solidity#8500]
I don't see why the following should not compile, but it does not:
```
pragma solidity 0.6.4;

interface A {
    function foo() external pure returns (string memory);
}

contract B is A {
    string public override foo = "bar";
}
```

Discovered by @sobolev-igor

**Comment by chriseth** [comment]
Maybe fixed by #8491 ?

**Comment by chriseth** [comment]
Yes, fixed by #8491
